grant-project/grant-monorepo
# S3 Avatar Signed Uploads

Closes #226.

### What this does
- BE provides signed avatar post for client
- FE uploads avatar to S3 using signed post
- BE removes avatars when appropriate
- BE generates a unique filename via UUID (it was MD5 hashing before, but now we are not uploading the file to BE, just the mimetype)
- `Avatar` model uses `hybrid_property` to limit stored data to file name, extracting filename on `set` and constructing full url on `get`
- adds `boto3` aws sdk dependency to BE

### Testing
- install BE deps
- follow S3 bucket & user directions in `README`
- edit a profile, add a photo, change the photo, delete the photo -- ensure the photo renders on the page
- the bucket should only have one avatar per user, and no avatar if its been deleted
